test/model/toRestHandlers.test.ts

```diff
@@ -126,10 +141,72 @@ describe('GET /users', () => {
       {
         id: 'ghi-789',
         firstName: 'Joseph',
+        lastName: 'Sipes',
       },
       {
         id: 'xyz-321',
         firstName: 'Eva',
+        lastName: 'Grant',
+      },
+    ])
+  })
+
+  it('return filtered entities', async () => {
+    server.use(...db.user.toHandlers('rest', 'http://localhost'))
+    db.user.create({
+      id: 'abc-123',
+      firstName: 'John',
+      lastName: 'White',
+    })
+    db.user.create({
+      id: 'def-456',
+      firstName: 'Kate',
+      lastName: 'Moen',
+    })
+    db.user.create({
+      id: 'def-789',
+      firstName: 'Kate',
+      lastName: 'Hilll',
+    })
+    const res = await fetch(
+      'http://localhost/users?firstName=Kate&lastName=Moen',
+    )
+    const users = await res.json()
+
+    expect(users).toEqual([
+      {
+        id: 'def-456',
+        firstName: 'Kate',
+        lastName: 'Moen',
+      },
+    ])
+  })
+
+  it('return all entities when wrong filter param is provided', async () => {
```

Comment:
Shouldn't it return nothing, as in query didn't find any results?

Reply:
How I have thought to skip those parameters, checking if the param is in the model definition. 
We can return an error in this case ?

Reply:
I have looked around and seams that 400 bad request is the best way for this use case. What do you think ?

Reply:
I think that's the right way to go!

I'd rephrase the error message to something like:

```
`Failed to query the "${modelName}" model: unknown property "${key}".`
```

Not sure about extending the `OperationErrorType` enum, though. It's meant to represent internal operation errors, while the wrong query parameter is a handler-specific case. I'd throw a regular error and mark it with a 400 status code:

```js
return res(ctx.status(400), ctx.json({ errorMessage: '...' }))
```

Do you think that would break the flow of how we handle the errors at the moment?

Reply:
To make error logic consistent I have extended the OperationError. In this way we can handle other errors in the `withErrors` funciton. Let me know what you think